app/javascript/hooks/useLinks.ts

```diff
@@ -28,12 +30,22 @@ export const useLinks = () => {
   );
 
   const handleMentionClick = useCallback(
-    (element: HTMLAnchorElement) => {
-      dispatch(
-        openURL(element.href, history, () => {
+    async (element: HTMLAnchorElement) => {
+      const result = await dispatch(openURL({ url: element.href }));
+
+      if (isFulfilled(result)) {
+        if (result.payload.accounts[0]) {
+          history.push(`/@${result.payload.accounts[0].acct}`);
+        } else if (result.payload.statuses[0]) {
+          history.push(
+            `/@${result.payload.statuses[0].account.acct}/${result.payload.statuses[0].id}`,
+          );
+        } else {
           window.location.href = element.href;
-        }),
-      );
+        }
+      } else if (isRejected(result)) {
+        window.location.href = element.href;
+      }
```

Comment:
This is not new, this is something that I failed to notice when it was introduced, but this causes a very confusing behavior: if those links fail to resolve, clicking them will cause the target to open *in the current tab*, replacing the web interface, which is unlike every other external link in the interface.

Reply:
Unfortunately, because the execution here is async, using `window.open` would trigger the pop-up blocker. Is that an acceptable trade-off in this scenario?

Reply:
I think this is possibly slightly better than having the current tab replaced, but it's indeed not great either way…